sdk/src/test/kotlin/io/provenance/scope/sdk/ClientTest.kt

```diff
@@ -28,18 +41,144 @@ class ClientTest : WordSpec() {
 
                 session.dataAccessKeys shouldContain encryptionKeyPair.public
             }
+            "set new session for non existing scope" {
+
+                val signingKeyPair = ProvenanceKeyGenerator.generateKeyPair()
+                val encryptionKeyPair = ProvenanceKeyGenerator.generateKeyPair()
+
+                val client = Client(
+                    SharedClient(ClientConfig(0, 0, 0, URI.create("http://localhost:5000"), mainNet = false)),
+                    Affiliate(
+                        DirectKeyRef(signingKeyPair.public, signingKeyPair.private),
+                        DirectKeyRef(encryptionKeyPair.public, encryptionKeyPair.private),
+                        Specifications.PartyType.OWNER
+                    )
+                )
+
+                client.inner.affiliateRepository.addAffiliate(
+                    signingPublicKey = signingKeyPair.public,
+                    encryptionPublicKey = encryptionKeyPair.public
+                )
+
+                val session =
+                    client.newSession(TestContract::class.java, TestContractScopeSpecificationDefinition::class.java)
+
+                session.spec!!.definition.name shouldBe "TestContract"
+                session.spec!!.functionSpecsList[0].funcName shouldBe "printTest"
+                session.spec!!.functionSpecsList[0].outputSpec.spec.name shouldBe "testRecord"
+            }
+        }
+        "Client.hydrate" should {
+            val signingKeyPair = ProvenanceKeyGenerator.generateKeyPair()
+            val encryptionKeyPair = ProvenanceKeyGenerator.generateKeyPair()
+
+            val scope = createExistingScope().apply {
+                scopeBuilder.scopeBuilder
+                    .clearDataAccess()
+                    .addDataAccess(encryptionKeyPair.public.getAddress(false))
+            }
+
+            mockkConstructor(CachedOsClient::class)
+            every { anyConstructed<CachedOsClient>().getRecord(any(), any(), any()) } returns
+                    Futures.immediateFuture(
+                        HelloWorldExample.ExampleName.newBuilder().setFirstName("Test").setLastName("TestLast").build()
+                    )
+            val client = Client(
+                SharedClient(ClientConfig(0, 0, 0, URI.create("http://localhost:5000"), mainNet = false)),
+                Affiliate(
+                    DirectKeyRef(signingKeyPair.public, signingKeyPair.private),
+                    DirectKeyRef(encryptionKeyPair.public, encryptionKeyPair.private),
+                    Specifications.PartyType.OWNER
+                )
+            )
+
+            client.inner.affiliateRepository.addAffiliate(
+                signingPublicKey = signingKeyPair.public,
+                encryptionPublicKey = encryptionKeyPair.public
+            )
+
+            shouldNotThrowAny {
+                client.hydrate(HelloWorldData::class.java, scope.build())
```

Comment:
probably would be good to assert some things off of the result of hydration (i.e. that HelloWorldData.name contains the values that CachedOsClient is mocked to return)

Reply:
Will do!